packages/as-types/assembly/serializable.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Implement this interface so that you can use `Args` to serialize/de-serialize your objects.
+ *
+ * @example
+ * ```typescript
+ * const args = new Args().addObject(new Address(input));
+ * const deserialized = args.nextObject<Address>(new Address());
+ * ```
+ */
+export interface Serializable<T> {
+  /**
+   * @returns the bytes
+   */
+  serialize(): StaticArray<u8>;
+
+  /**
+   * @param data - the serialized data
+   * @param offset - current offset
+   * @returns the length of the deserialized object
+   */
+  deserialize(data: StaticArray<u8>, offset: i32): i32;
```

Comment:
How do you plan to tell the caller that deserialization went bad ?

Do you enforce the panic mode ? Use a Result ? Return the same value of the offset ? return 0 ? -1 ?

Reply:
it now returns a Result